python/google/protobuf/json_format.py

```diff
@@ -631,6 +631,9 @@ def _ConvertGenericMessage(self, value, message):
     """Convert a JSON representation into message with FromJsonString."""
     # Duration, Timestamp, FieldMask have a FromJsonString method to do the
     # conversion. Users can also call the method directly.
+    if not isinstance(value, str):
```

Comment:
In this fix I was following the precedent of type-checking that's performed in `_ConvertListValueMessage` in this same class.

Performing a type check in `FromJsonString`  is valid but:
1. It means adding code to each WKT class.
2. Adding it to FromJsonString would handle the case where the user calls FromJsonString directly, but in that case it's easier to diagnose the error. What I'm addressing here is when parsing a larger JSON into a Protobuf results in one of the fields failing to parse. At that point it's hard to tell from the existing behavior which field failed to parse and for what reason.

Reply:
"following _Parser._ConvertListValueMessage."
_ConvertListValueMessage does not call any public API. But FromJsonString() is a public API we will call here.

"It's hard to tell from the existing behavior which field failed to parse and for what reason." 
Raise ValueError in FromJsonString, it will be caught in this function (by try... except.. )and is able to tell which field failed.

Reply:
You have a point.